src/NuGet.Clients/NuGet.PackageManagement.UI/DisplayVersion.cs

```diff
@@ -79,6 +79,58 @@ public class DisplayVersion
             }
         }
 
+        public DisplayVersion(
```

Comment:
This class already has a constructor, and this method looks like it's duplicating a lot of the code. What's better is to have one constructor call the other one:

```csharp
class MyClass
{
  public MyClass(string arg)
    : this(arg, null)
  {
  }

  public MyClass(string arg, object thing)
  {
    // class constructor code here
  }
}
```

This way there's only one constructor with code in its body, which means that there's less risk of bugs if someone modifies one of the constructors to do something different, but doesn't change the other one.


However, having written all that, you should also consider if it's worthwhile (risk of bugs, technical debt) to keep both constructors, or just change all usage to use the new constructor instead, and delete the old constructor. Unless a method/constructor is in a PublicAPI.Shipped.txt file, it's best to delete anything unused, and to generally optimize for long term maintainability.